src/controller/CHIPDeviceController.h

```diff
@@ -367,6 +375,32 @@ class DLL_EXPORT DeviceCommissioner : public DeviceController, public Rendezvous
      */
     CHIP_ERROR CloseBleConnection();
 #endif
+    /**
+     * @brief
+     *   Discover devices advertising as commissionable that match the long discriminator.
+     * @return CHIP_ERROR   The return status
+     */
+    CHIP_ERROR DiscoverCommissioningLongDiscriminator(uint16_t long_discriminator);
+
+    /**
+     * @brief
+     *   Discover all devices advertising as commissionable.
+     * @return CHIP_ERROR   The return status
+     */
+    CHIP_ERROR DiscoverAllCommissioning();
+
+    /**
+     * @brief
+     *   Helper function to print device info for device selection.
```

Comment:
Could this be removed from this public API and replaced by example code doing same thing at call site based on GetDiscoveredDevice ?

Reply:
I chose to do it this way because I didn't want random prints showing up in the middle console output if new mdns packets arrive during operation. 

If you object to the print, I could also move the printing to the python side.

Reply:
Yes, strongly suggest not having printers in public API and letting clients print.